microkernel-architecture/src/main/java/com/iluwatar/microkernel/adapters/Adapter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The MIT License
+ * Copyright © 2014-2019 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwatar.microkernel.adapters;
+
+import com.iluwatar.microkernel.externals.ResultProcessorServer;
+import com.iluwatar.microkernel.microkernel.BudgetMicrokernel;
+
+/**
+ * This class represents the adapter.
+ * This class hides system dependencies such as
+ * communication facilities from the client,
+ * invokes methods of external servers on behalf of clients.
+ */
+public class Adapter {
+
+  private BudgetMicrokernel microkernel;

```

Comment:
Use final keyword

Reply:
Check this elsewhere too. Make the members final if they are not modified after the initialization.

Reply:
@iluwatar Is it a good idea to request authors as part of PR template to sonarqube lint tools to verify locally for avoid critical issues? This gives opportunity for the developer to look at smaller issues as well.

Reply:
In principle yes, but I'm afraid it makes creating pull requests more complicated. It would be great if the CI build could check for Sonarqube issues in the incoming changed code. Not sure if Sonarcloud can be configured to do that..